commands/commands.go

```diff
@@ -384,15 +384,19 @@ func CmdAuth(c *cli.Context) error {
 		}
 	}
 
-	fmt.Print("Password: ")
-	pass, err := gopass.GetPasswdMasked()
-	if err != nil {
-		return cli.NewExitError(fmt.Sprintf("error reading password: %v", err), 1)
-	}
-
-	// Validate password
+	// Take password from argument, and fall back to input
+	pass := c.String("p")
 	if len(pass) == 0 {
-		return cli.NewExitError("Please enter your password.", 1)
+		fmt.Print("Password: ")
+		pass, err := gopass.GetPasswdMasked()
```

Comment:
Since `pass` is getting re-declared here, it will only be local to the encompassing `if` statement and won't make it into the `api.DoLogIn()` call below, causing interactive password entry to break.

I'd recommend declaring a new var here instead (since `gopass.GetPasswdMasked()` returns a `[]byte`), and then doing 
```
pass = string(thatNewVar)
```

Reply:
Tried that, but it's not accepting the parameter.

Reply:
Nevermind, was editing the wrong checkout.